tst/providers/MeetingProvider/index.test.tsx

```diff
@@ -0,0 +1,79 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import {
+  LogLevel,
+  EventName, 
+  EventAttributes, 
+  MeetingSessionCredentials, 
+  MeetingSessionURLs
+} from 'amazon-chime-sdk-js';
+
+import MeetingManager from '../../../src/providers/MeetingProvider/MeetingManager';
+import { MeetingJoinData } from '../../../src/providers/MeetingProvider/types';
+
+
+describe('Meeting Provider', () => {
+
+  it('events are received correctly', async (done) => {
+    // Mock the user agent to ensure the MeetingManager event subscription is 
+    // set up. Otherwise, the amazon-chime-sdk-js will not detect a valid
+    // browser to use.
+    const userAgentGet = jest.spyOn(navigator, 'userAgent', 'get');
+    userAgentGet.mockReturnValue('Chrome/96.0');
+    
+    // Event details
+    const eventName = 'audioInputFailed';
+    const audioInputErrorMessage = 'Something went wrong';
+
+    // Setup meeting manager and eventcontroller
+    let meetingManager = new MeetingManager({ logLevel: LogLevel.OFF});
+    let joinData: MeetingJoinData = {
+      meetingInfo: {
+        meetingId: '',
+        externalMeetingId: '',
+        mediaplacement: new MeetingSessionURLs()
+      }, 
+      attendeeInfo: new MeetingSessionCredentials(),  
+    }
+    await meetingManager.join(joinData);
+
+    let calls = 0;
+    let callback1 = (name: EventName, attributes: EventAttributes) => {
+      expect(name).toBe(eventName);
+      expect(attributes.audioInputErrorMessage).toBe(audioInputErrorMessage);
+      calls += 1;
+      // Should only get called once
+      expect(calls).toBe(1);
+    };
+    let callback2 = (name: EventName, attributes: EventAttributes) => {
+      expect(name).toBe(eventName);
+      expect(attributes.audioInputErrorMessage).toBe(audioInputErrorMessage);
+      done();
```

Comment:
done is unnecessary.

Reply:
We need the done to ensure the callback is actually being called

Reply:
dont we already have await setTimeout to wait for callback?

Reply:
So that wouldn't work before, but I rewrote the logic to make it work without done